src/shared/UniversalRender.jsx

```diff
@@ -26,10 +26,63 @@ import extractMeta from 'app/utils/ExtractMeta';
 import Translator from 'app/Translator';
 import {notificationsArrayToMap} from 'app/utils/Notifications';
 import {routeRegex} from "app/ResolveRoute";
-import {contentStats} from 'app/utils/StateFunctions'
+import {contentStats} from 'app/utils/StateFunctions';
+import ScrollBehavior from 'scroll-behavior';
 
 import {api} from 'steem';
 
+
+const calcOffsetRoot = (startEl) => {
+    let offset = 0;
+    let el = startEl;
+    while(el) {
+        offset += el.offsetTop;
+        el = el.offsetParent;
+    }
+    return offset;
+}
+
+//BEGIN: SCROLL CODE
+const SCROLL_TOP_TRIES = 20;
+const SCROLL_TOP_DELAY_MS = 50;
```

Comment:
If I understand this right, the code will try 20 times over a period of 1 second to scroll to the desired element. I'm concerned this is not enough time.

Reply:
I'd like to err on the conservative side. If our pages take a more than a second to render we should be doing something about that.

Reply:
Our pages take well over 1s to load, and when there are lots of images they may be loading for quite some time as well. We can leave as-is for now but let's see how it works in practice; I have a feeling 5-10s may be necessary. I doubt a user would want to scroll anywhere else that fast if they're jumping to a specific post*. If we could detect that all images are loaded we can stop trying early.

* but does this code also affect e.g. scrolling to `#comments`?

Reply:
<3

It affects urls with `#comments` as the hash component. That's actually how the delay requirement was discovered. I was getting partial scrolls before all the child comments had loaded (and other odd behavior)